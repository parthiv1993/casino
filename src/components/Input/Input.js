import PropTypes from 'prop-types';
import Icon from '../Icon';
import { StyledInput, StyledInputContainer } from './Input.style';

const Input = ({ suffixIcon, onChange, ...restProps }) => {
  return (
    <StyledInputContainer>
      <StyledInput
        {...restProps}
        onChange={(e) => {
          onChange(e.target.value);
        }}
      />
      {suffixIcon && <Icon iconName={suffixIcon} />}
    </StyledInputContainer>
  );
};

Input.propTypes = {
  suffixIcon: PropTypes.string,
  onChange: PropTypes.func
};

Input.defaultProps = {
  suffixIcon: null,
  onChange: () => {}
};

export default Input;
