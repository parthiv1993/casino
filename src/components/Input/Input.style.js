import styled from 'styled-components';

export const StyledInput = styled.input`
  border: 0;
  outline: none;
  padding: 0;
  flex: 1;
  background: transparent;
`;

export const StyledInputContainer = styled.div`
  width: 100%;
  ${({ theme }) => `
        border: 1px solid ${theme.Input.borderColor};
        background-color: ${theme.Input.background};
        &&:focus-within {
          outline: 2px solid ${theme.Input.outline};
        }
    `};
  border-radius: 4px;
  display: flex;
  gap: 0.5em;
  padding: 0.5em;
`;
