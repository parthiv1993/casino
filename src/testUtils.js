import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import light from './contexts/theme/light';

export const renderComponentWithLightTheme = (children) => {
  return render(<ThemeProvider theme={light}>{children}</ThemeProvider>);
};

export const lowerCaseAllKeys = (props = {}) => {
  const lowercaseObject = {};
  Object.keys(props).map((key) => (lowercaseObject[key.toLowerCase()] = props[key]));

  return lowercaseObject;
};

export const MockIcon = (props) => <div component="Icon" {...lowerCaseAllKeys(props)} />;
