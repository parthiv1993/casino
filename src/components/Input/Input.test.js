import { fireEvent } from '@testing-library/react';
import { renderComponentWithLightTheme } from '../../testUtils';
import Input from './Input';

describe('Input', () => {
  it('should match snapShot when there is no suffic Icon', () => {
    const { container } = renderComponentWithLightTheme(<Input />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapShot when there is suffic Icon', () => {
    const { container } = renderComponentWithLightTheme(<Input suffixIcon={'fa-username'} />);
    expect(container).toMatchSnapshot();
  });

  it('should pass the props', () => {
    const { getByTestId } = renderComponentWithLightTheme(
      <Input suffixIcon={'fa-username'} type={'password'} data-testid="test-input" />
    );
    expect(getByTestId('test-input')).toHaveProperty('type', 'password');
  });

  it('should trigger onChange event', () => {
    const mockOnChange = jest.fn();
    const { getByTestId } = renderComponentWithLightTheme(
      <Input
        suffixIcon={'fa-username'}
        type={'password'}
        data-testid="test-input"
        onChange={mockOnChange}
      />
    );
    fireEvent.change(getByTestId('test-input'), { target: { value: 'abc' } });
    expect(mockOnChange).toHaveBeenCalledTimes(1);
    expect(mockOnChange).toHaveBeenCalledWith('abc');
  });
});
