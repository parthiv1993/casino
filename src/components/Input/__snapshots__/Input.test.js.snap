// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapShot when there is no suffic Icon 1`] = `
.c1 {
  border: 0;
  outline: none;
  padding: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: transparent;
}

.c0 {
  width: 100%;
  border: 1px solid rgba(0,0,0,0.15);
  background-color: white;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5em;
  padding: 0.5em;
}

.c0.c0:focus-within {
  outline: 2px solid blue;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
    />
  </div>
</div>
`;

exports[`Input should match snapShot when there is suffic Icon 1`] = `
.c1 {
  border: 0;
  outline: none;
  padding: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: transparent;
}

.c0 {
  width: 100%;
  border: 1px solid rgba(0,0,0,0.15);
  background-color: white;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5em;
  padding: 0.5em;
}

.c0.c0:focus-within {
  outline: 2px solid blue;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
    />
    <i
      class="fa-username"
    />
  </div>
</div>
`;
